packages/compile-solidity/index.js

```diff
@@ -74,6 +46,10 @@ const Compile = {
 
   // this takes an array of paths and options
   async sourcesWithDependencies({ paths, options }) {
+    if (options.compilers.solc.version === "analyzePragmas") {
+      return await this.sourcesWithPragmaAnalysis({ paths, options });
```

Comment:
Yeah, I realize that using the `await` is unnecessary (I didn't know it pushed it to the end of the queue). I kind of wrestle with the idea of intentionally being explicit about the fact that it is an asynchronous function. I didn't really know that it makes a real difference in performance. I don't really feel strongly about it so I'll be happy to omit this in the future.

Reply:
I'd say that marking this function as `async` is explicit enough here (although doing that _also_ sends the return of the promise to end of the queue*). That said, I'm actually wrong*; applying `async` to a function that returns a promise is what bumps it to the back of the promise queue, `return await` is just redundant here and has no effect on performance (though the `async` keyword does).

I _kinda_ do like the `async` keyword here, so I'm torn as to what we should do here, so I'll side with what I think gnidan would prefer... which is to keep the keyword (and `await`, since it doesn't matter). You can mark this and the other related review comments as resolved if you agree. :-)

* Why I think I was wrong:

```javascript
async function fn(r){
  return Promise.resolve(r);
}
async function a(){
  return await fn("a");
}
async function b(){
  return fn("b");
}

// which ever is called first wins
await Promise.race([a(), b()]) // "a"
await Promise.race([b(), a()]) // "b"
```

Reply:
All I did so far was get rid of the superfluous `await`s. Is that good?

Reply:
I think so